handbook/engineering/search/index.md

```diff
@@ -2,9 +2,99 @@
+
+- **Search Expressions (AND/OR/NOT)** ([term explanation](https://github.com/sourcegraph/sourcegraph/issues/13126), [term RFC](https://docs.google.com/document/d/1SHky6nodPs1w_zRXz24jB2nq5LbMCJl1u7hcLTGHDL8/edit#))
+  - **Problem:** We have customers who want to migrate from OpenGrok and other search tools, and they want to be able to do searches that are available in those tools.
+  - **Outcomes:** OpenGrok users eagerly migrate to Sourcegraph, and can run complex searches on Sourcegraph.
+  - **Owner:** Rijnard
+  - **Status:** In progress
+- **Improve syntax for existing filters like `repohasfile`**
+  - **Problem:** Creating a custom filter name for every permutation of search use does not scale (existing filter examples: `repohasfile`, `hascommitafter`).
+  - **Outcomes:** We have an expressive syntax that scales and effectively leverages existing filters to achieve searches that previously would have required a custom/new filter.
+  - **Owner:** Rijnard/Stefan
+  - **Status:** Not started
+- **Revision search**
+  - **Problem:** Users want to search across branches with the same ease as searching across repositories, and are currently unable to do so.
+  - **Owner:** Stefan
+  - **Status:** In progress
+
+### Easy to use
+
+- **Get to files and repositories more quickly**
+  - **Problem:** Finding files and repositories often takes multiple tries to get the syntax correct or creative solutions to get the result users are looking for.
+  - **Plan:** Wildcard syntax
+  - **Outcomes:** Users leverage file and repo filters to quickly get to the code they are searching for.
+  - **Owner:** Stefan
+  - **Status:** In progress
+- **Improve search experience**
+  - **Problem:**
+      - New users who are trying Sourcegraph for the first time have trouble learning the syntax and breadth of Sourcegraph features.
+          - **Plan:** Search onboarding tour
+          - **Owner:** Farhan
+          - **Status:** In progress
+      - It is hard for users to quickly get to code they care about.
+          - **Plan:** Enterprise homepage
+          - **Owner:** Farhan, Juliana
+          - **Status:** In progress
+    - **Outcomes:**
+      - New users introduced to Sourcegraph are able to more run searches and see the value of Sourcegraph more quickly.
+      - Users can run searches over code they care about more quickly.
+      - Sourcegraph surfaces code and searches users care about.
+
+### Shareable
+
+- **Code monitoring** ([RFC 227](https://docs.google.com/document/d/1_R5DgpUkxyZilsJ9vBQm5cvRPT2udc3tZIPg2q3cnZU/edit))
+  - **Problem:** Users want to be notified about important things going on in their code.
+  - **Outcome:** Notifications create a shared understanding and raise awareness of what’s going on in the code.
+  - **Plan**
+      - Improved diff search performance
+      - Webhooks on search results
+          - Slack integration
+          - Zapier integration
+          - Email
+  - **Owner:** TBD
+  - **Status:** Not started
```

Comment:
This short-term goal does not appear to tie back to the medium-term goal. Is this intentional?

Reply:
No - this isn’t intentional. The medium term goals are focussed on how search can be “fast” and “easy to use” right now as those are problems we anticipate needing to solve the most.